src/Datadog.Trace/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Linq;
+
+namespace Datadog.Trace.Plugins
+{
+    internal static class PluginManager
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(PluginManager));
+
+        public static IReadOnlyCollection<IOTelPlugin> Loaded { get; private set; } = ArrayHelper.Empty<IOTelPlugin>();
+
+        internal static void TryLoadPlugins(JsonConfigurationSource pluginsConfig)
+        {
+            if (pluginsConfig == null)
+            {
+                return;
+            }
+
+            var runtimeName = GetPluginsRuntimeName();
+
+            Log.Debug("Executing plugins configuration: {0}", pluginsConfig);
+            Log.Information("Trying to load plugins for '{0}' runtime.", runtimeName);
+
+            // TODO: Detect if objects instead of path
+            var pluginFiles = pluginsConfig?.GetValue<JToken>($"['{runtimeName}']").ToObject<string[]>();
+
+            if (pluginFiles == null || !pluginFiles.Any())
+            {
+                Log.Information("Skipping plugins load. Could not find any plugins for '{0}' runtime.", runtimeName);
+
+                return;
+            }
+
+            var loadedPlugins = TryLoadPlugins(pluginFiles);
+
+            Log.Information("Successfully loaded '{0}' plugin(s).", property: loadedPlugins.Count);
+
+            Loaded = loadedPlugins;
+        }
+
+        private static IReadOnlyCollection<IOTelPlugin> TryLoadPlugins(string[] pluginFiles)
+        {
+            var loaded = new List<IOTelPlugin>();
+
+            foreach (string file in pluginFiles)
+            {
+                string fullPath = Path.GetFullPath(file);
+
+                if (File.Exists(fullPath))
```

Comment:
I really think it is not needed.
`Assembly.LoadFrom()` should throw `FileNotFoundException` if the file is missing.
There is always a non-zero possibility for a "race" when the file exists here and does not exist when you load the assembly. So I think it is enough to handle this case only during assembly loading

Reply:
Currently it's a performance optimization (if vs exception). Caused of defaults in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/129#pullrequestreview-647515612